homeassistant/components/jewish_calendar/config_flow.py

```diff
@@ -0,0 +1,132 @@
+"""Config flow for Jewish calendar integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+import zoneinfo
+
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LANGUAGE,
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_TIME_ZONE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    LocationSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "jewish_calendar"
+CONF_DIASPORA = "diaspora"
+CONF_CANDLE_LIGHT_MINUTES = "candle_lighting_minutes_before_sunset"
+CONF_HAVDALAH_OFFSET_MINUTES = "havdalah_minutes_after_sunset"
+DEFAULT_NAME = "Jewish Calendar"
+DEFAULT_CANDLE_LIGHT = 18
+DEFAULT_DIASPORA = False
+DEFAULT_HAVDALAH_OFFSET_MINUTES = 0
+DEFAULT_LANGUAGE = "english"
+
+LANGUAGE = [
+    SelectOptionDict(value="hebrew", label="Hebrew"),
+    SelectOptionDict(value="english", label="English"),
+]
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_CANDLE_LIGHT_MINUTES, default=DEFAULT_CANDLE_LIGHT): int,
+        vol.Optional(
+            CONF_HAVDALAH_OFFSET_MINUTES, default=DEFAULT_HAVDALAH_OFFSET_MINUTES
+        ): int,
+    }
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_data_schema(hass: HomeAssistant) -> vol.Schema:
+    default_location = {
+        CONF_LATITUDE: hass.config.latitude,
+        CONF_LONGITUDE: hass.config.longitude,
+    }
+    return vol.Schema(
+        {
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+            vol.Required(CONF_DIASPORA, default=DEFAULT_DIASPORA): BooleanSelector(),
+            vol.Required(CONF_LANGUAGE, default=DEFAULT_LANGUAGE): SelectSelector(
+                SelectSelectorConfig(options=LANGUAGE)
+            ),
+            vol.Required(CONF_LOCATION, default=default_location): LocationSelector(),
+            vol.Optional(CONF_ELEVATION, default=hass.config.elevation): int,
+            vol.Optional(CONF_TIME_ZONE, default=hass.config.time_zone): SelectSelector(
+                SelectSelectorConfig(
+                    options=sorted(zoneinfo.available_timezones()),
+                )
+            ),
```

Comment:
Why does timezone matter?

Reply:
The library needs to know the timezone to calculate the times correctly

Reply:
But why is it customizable? We can get the timezone from the hass object right?

Reply:
No, if the user decides to specify their own location for this integration, they should provide the timezone for tthat location.
It might be somewhere else entirely.

Reply:
Please add `data_description` to strings.json to better explain these options are for customizing the location used to calculate the calendar times. The new posibilities added by https://github.com/home-assistant/core/pull/116100 would probably be great (if extended to also cover data entry flows).

Reply:
I think this has been addressed @joostlek